enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/expressions/NodeProperty.scala

```diff
@@ -34,3 +34,20 @@ case class NodeProperty(offset: Int, token: Int) extends Expression {
 
   override def symbolTableDependencies: Set[String] = Set.empty
 }
+
+case class NodePropertyLate(offset: Int, propKey: String) extends Expression {
+
+  override def apply(ctx: ExecutionContext)(implicit state: QueryState): Any = {
+    val maybeToken = state.query.getOptPropertyKeyId(propKey)
+    if (maybeToken.isEmpty)
```

Comment:
@systay @craigtaverner why no remove the `QueryContext#getOpt*()` methods altogether, and replace with `QueryContext#has*()` + `QueryContext#get*()`?